### PR TITLE
Add some unix-only cram tests to test for #4216 before fixing it

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Will help to merge #4901

### CODE REVIEW

**Comment by rjbou** [comment]
is there a reason to not use reftest engine (`run.ml`) ?

**Comment by kit-ty-kate** [comment]
I needed support for actual POSIX shell syntax (which `run.ml` does not allow) and the bultin dune cram tests are much simpler to use too. Since we don’t need windows support for those specific tests, it’s much better to simply use the bultin cram tests.

File: master_changes.md [hunk 1]
```diff
@@ -212,6 +213,8 @@ users)
   * Escape regexps characters in string replacements primitives [#5009 @kit-ty-kate]
   * Automatically update default repo when adding a package file [#5004 @AltGr]
 
+## Cram tests
+  * Add some unix-only cram tests to test for https://github.com/ocaml/opam/issues/4216 before fixing it [#5022 @kit-ty-kate]
```
Comment (rjbou) [inline master_changes.md:217]: Only if a re run / update pr is done

```suggestion
### Cram tests
  * Add some unix-only cram tests to test for https://github.com/ocaml/opam/issues/4216 before fixing it [#5022 @kit-ty-kate]
```

File: tests/cram/unix/list.t [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Test https://github.com/ocaml/opam/issues/4216
+  $ export OPAMROOT=./opamroot
+  $ opam init --bare -n git+https://github.com/ocaml/opam-repository.git#0ec9bbf73a91b8e90b15bb11735859c9552a9888 > /dev/null
+  $ OCAMLRUNPARAM=b opam list --depends-on dune -s --all-versions | head -n1
+  0install.2.14
+  Fatal error: exception Sys_error("Broken pipe")
+  Raised by primitive operation at OpamCliMain.main_catch_all in file "src/client/opamCliMain.ml", line 357, characters 4-16
+  Called from Dune__exe__OpamMain in file "src/client/opamMain.ml", line 12, characters 9-28
```
Comment (rjbou) [inline tests/cram/unix/list.t:8]: locally test passes with these modifications

```suggestion
  $ OPAMSWITCH= OCAMLRUNPARAM=b opam list --depends-on dune -s --all-versions | head -n1
  0install.2.14
  Fatal error: exception Sys_error("Broken pipe")
  Raised by primitive operation at Stdlib in file "stdlib.ml" (inlined), line 337, characters 0-54
  Called from OpamCliMain.main_catch_all in file "src/client/opamCliMain.ml", line 357, characters 4-16
  Called from OpamCliMain.main in file "src/client/opamCliMain.ml" (inlined), line 476, characters 2-8
  Called from Dune__exe__OpamMain in file "src/client/opamMain.ml", line 12, characters 9-28
```

**Comment by AltGr** [comment]
We're still not rid of opam-rt. **please** don't multiply the test frameworks unless there really is no other way. Think of the added maintainance cost: one more place to look every time for testing, different rules to run or upgrade the tests, etc.

**Comment by kit-ty-kate** [comment]
If we really want to demultiply the test frameworks we can do it at a later date. I’m responsible for it, I’m not going anywhere. Right now, this is moving us forward.

**Comment by AltGr** [comment]
I for one prefer to keep things clean in the first place rather than let others clean up after me; this could have least have been discussed.
Is https://github.com/ocaml/opam/pull/5031/commits/cd4815e3a3630ee7d73ce64c9003efc7b7d5c8e7 so overly complex to you ?

**Pull request merged**